pkg/game/board.go

```diff
@@ -0,0 +1,134 @@
+package game
+
+import (
+	"errors"
+	"slices"
+
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/game/elements"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tilesets"
+)
+
+// mutable type
+// Position coordinates example on the board:
+// (-1, +1)  (+0, +1)  (+1, +1)
+// (-1, +0)  (+0, +0)  (+1, +0)
+// (-1, -1)  (+0, -1)  (+1, -1)
+//
+// Starting tile is placed at (+0, +0) position.
+type board struct {
+	tileSet tilesets.TileSet
+	// The information about the tile and its placement is stored sparsely
+	// in a slice of size equal to the number of tiles in the set.
+	// `tiles[0]` is always the starting tile.
+	// Indexes in `tiles[1:]` correspond to positions of equivalent tiles in `tileSet`.
+	tiles []elements.PlacedTile
+	// tilesMap is used by the engine for faster lookups
+	// but contains the same information as the `tiles` slice.
+	tilesMap map[elements.Position]elements.PlacedTile
+}
+
+func NewBoard(tileSet tilesets.TileSet) elements.Board {
+	tiles := make([]elements.PlacedTile, len(tileSet.Tiles)+1)
+	startingTile := elements.NewStartingTile(tileSet)
+	tiles[0] = startingTile
+	return &board{
+		tileSet: tileSet,
+		tiles:   tiles,
+		tilesMap: map[elements.Position]elements.PlacedTile{
+			elements.NewPosition(0, 0): startingTile,
+		},
+	}
+}
+
+func (board *board) TileCount() int {
+	return len(board.tilesMap)
+}
+
+func (board *board) Tiles() []elements.PlacedTile {
+	return board.tiles
+}
+
+func (board *board) GetTileAt(pos elements.Position) (elements.PlacedTile, bool) {
```

Comment:
```suggestion
func (board *board) GetTileAtPosition(pos elements.Position) (elements.PlacedTile, bool) {
```
Just a suggestion

Reply:
I'd say that it's clear from the signature what `At` is referring to (i.e. calls are likely to be either `board.GetTileAt(pos)` or `board.GetTileAt(elements.NewPosition(...))` where it's clear that we're trying to get a tile at a specified position) and so there's no much point in duplicating that information in the method name.